Makefile

```diff
@@ -16,14 +16,17 @@ format:
 	$(DART_CLI) format $(FLUTTER_APP_ROOT)
 
 clean:
-	$(FLUTTER_CLI) clean --packages $(FLUTTER_APP_ROOT)
+	pushd $(FLUTTER_APP_ROOT) && $(FLUTTER_CLI) clean && popd
 	$(MAVEN_CLI) clean --file $(MAVEN_FRONTEND_ROOT)/pom.xml
 
-build-and-copy-flutter:
+flutter-pub-get:
+	pushd $(FLUTTER_APP_ROOT) && $(FLUTTER_CLI) pub get && popd
+
+build-and-copy-flutter: flutter-pub-get
 	pushd $(FLUTTER_APP_ROOT) && $(FLUTTER_CLI) build web && popd
 	cp -r $(FLUTTER_APP_ROOT)/build/web/* $(MAVEN_FRONTEND_ROOT)/src/main/webapp/
 
-test-flutter:
+test-flutter: flutter-pub-get
```

Comment:
This is a good idea, but `pub get` can be slow and expensive and doesn't generally need to be run on every execution.

Ideally, we run `pub get` whenever the YAML config updates but not otherwise. Unfortunately, we need to know that it has run at least once; we should be able to use a sentinel file to indicate that an update has run, but I wasn't able to get it working on macOS (GNU Make may support this correctly).

@KoopaKing Any suggestions?

Reply:
pub get is not expensive. It looks to see if the specified versions are already in the pub cache, which they will be unless clean has run.

Reply:
Ack